include/circt/Dialect/Arc/ArcOps.td

```diff
@@ -800,4 +800,21 @@ def VectorizeReturnOp : ArcOp<"vectorize.return", [
   let assemblyFormat = "operands attr-dict `:` qualified(type(operands))";
 }
 
+def AssertOp : ArcOp<"assert"> {
+  let summary = "assert that a properity holds.";
+  let description = [{
+    Asserts that an operation holds at runtime, it takes a single boolean
+    as an input, and an error message.
+    If the property holds (= `true`) the operation has no effect. Otherwise, the
+    execution aborts and provide an error message to the user.
+    ```mlir
+    assert %proberty, "Some error message in case of failure!"
+    ```
+  }];
+  // I thought of making the string message optional.
```

Comment:
Sounds reasonable since it is also optional in `verif.assert`. Would you otherwise lower it to the empty string if it is not present in `verif.assert`?